patches/minecraft/net/minecraft/client/renderer/ItemRenderer.java.patch

```diff
@@ -56,21 +56,44 @@
                  this.func_178108_a(p_78447_1_, this.field_78455_a.func_175602_ab().func_175023_a().func_178122_a(iblockstate));
              }
          }
-@@ -465,11 +470,13 @@
+@@ -465,11 +470,19 @@
          {
              if (this.field_78455_a.field_71439_g.func_70055_a(Material.field_151586_h))
              {
+-                this.func_78448_c(p_78447_1_);
 +                if (!net.minecraftforge.event.ForgeEventFactory.renderWaterOverlay(field_78455_a.field_71439_g, p_78447_1_))
-                 this.func_78448_c(p_78447_1_);
++                this.renderWaterOverlayTexture(p_78447_1_, field_110929_d);
              }
++            Block viewBlock = ActiveRenderInfo.func_186703_a(this.field_78455_a.field_71441_e, this.field_78455_a.func_175606_aa(), p_78447_1_).func_177230_c();
++            if(viewBlock instanceof net.minecraftforge.fluids.IFluidBlock)
++            {
++            	ResourceLocation overlay = ((net.minecraftforge.fluids.IFluidBlock)viewBlock).getOverlayTexture();
++                if(overlay != null) this.renderWaterOverlayTexture(p_78447_1_, overlay);
++            }
  
              if (this.field_78455_a.field_71439_g.func_70027_ad())
              {
 +                if (!net.minecraftforge.event.ForgeEventFactory.renderFireOverlay(field_78455_a.field_71439_g, p_78447_1_))
                  this.func_78442_d(p_78447_1_);
              }
          }
-@@ -591,8 +598,8 @@
+@@ -504,12 +517,12 @@
+         GlStateManager.func_179131_c(1.0F, 1.0F, 1.0F, 1.0F);
+     }
+ 
+-    private void func_78448_c(float p_78448_1_)
```

Comment:
I think removing vanilla methods like this is not allowed, it probably makes patch migration confusing.
You can replace the body with ` this.renderWaterOverlayTexture(p_78448_1_, field_110929_d);` though, which would mean you don't need to replace the call to it earlier in the patch.